csrc/quantization/cutlass_w8a8/scaled_mm_c3x.cu

```diff
@@ -125,6 +139,50 @@ struct ScaledEpilogue {
   }
 };
 
+template <typename ElementAcc, typename ElementD, typename EpilogueDescriptor>
+struct ScaledEpilogueBias
+    : private ScaledEpilogueBase<ElementAcc, ElementD, EpilogueDescriptor> {
+ private:
+  using SUPER = ScaledEpilogueBase<ElementAcc, ElementD, EpilogueDescriptor>;
+  using Accum = typename SUPER::Accum;
+  using ScaleA = typename SUPER::ScaleA;
+  using ScaleB = typename SUPER::ScaleB;
+
+  using Compute0 = cutlass::epilogue::fusion::Sm90Compute<
+      cutlass::multiplies, float, float,
+      cutlass::FloatRoundStyle::round_to_nearest>;
+
+  using EVTCompute0 =
+      cutlass::epilogue::fusion::Sm90EVT<Compute0, ScaleB, Accum>;
+
+  using Compute1 = cutlass::epilogue::fusion::Sm90Compute<
+      cutlass::multiply_add, ElementD, float,
+      cutlass::FloatRoundStyle::round_to_nearest>;
+
+  using Bias = cutlass::epilogue::fusion::Sm90RowBroadcast<
+      0 /*Stages*/, typename EpilogueDescriptor::TileShape, float,
+      Stride<Int<0>, Int<1>, Int<0>>, 128 / sizeof_bits_v<float>, false>;
+
+ public:
+  using EVTCompute =
+      cutlass::epilogue::fusion::Sm90EVT<Compute1, ScaleA, EVTCompute0, Bias>;
+  using ArgumentType = typename EVTCompute::Arguments;
+
+  static ArgumentType prepare_args(torch::Tensor const& a_scales,
+                                   torch::Tensor const& b_scales,
+                                   torch::Tensor const& bias) {
+    using ScaleA_Args = typename ScaleA::Arguments;
+    using ScaleB_Args = typename ScaleB::Arguments;
+    using Bias_Args = typename Bias::Arguments;
+
+    ScaleA_Args a_args{a_scales.data_ptr<float>(), a_scales.numel() != 1, {}};
+    ScaleB_Args b_args{b_scales.data_ptr<float>(), b_scales.numel() != 1, {}};
+    Bias_Args bias_args{bias.data_ptr<float>()};
+
+    return ArgumentType{a_args, {b_args}, bias_args};
```

Comment:
It'd be nice if there's some documentation about why the ArgumentType is structured this way. When I implement this, I only find examples in cutlass code and tried to go with this:
```
  args.epilogue.thread = {
      // ternary op: a_scales * (b_scales * (A*B)) + per-row bias
      a_args,  // a_scales
      {
          // binary op: b_scales * (A*B)
          b_args,  // b_scales
          {},      // acc
          {}       // binary args: multiplies
      },
      bias_args,  // bias
      {}          // ternary args: multiply_add
  };
```
I suppose there is some documentation about what can be omitted?

Reply:
They are just structs, like [this one](https://github.com/NVIDIA/cutlass/blob/637b15906358191cb4238af419d408a65819d7ec/include/cutlass/epilogue/fusion/sm90_visitor_load_tma_warpspecialized.hpp#L568-L571) (used for bias):
```
  struct Arguments {
    Element const* ptr_row = nullptr;
    Element null_default = Element(0);
    StrideMNL dRow = {};
  };
```

So they all have default arguments. For compute nodes, the arguments are a tuple of arguments of children.

I am inclined to keep it this way for conciseness as I don't think it's as complicated. But I'll defer to @tlrmchlsmth here.

Reply:
I think I actually did a slightly better job of breaking it down in the c2x kernel than in the c3x kernel: https://github.com/vllm-project/vllm/blob/a7dcc62086ea751b46b4821c2811cf8ac83711bf/csrc/quantization/cutlass_w8a8/scaled_mm_c2x.cu#L130-L133

Whitespace could be improved there, but that code is more explicit about what the types are, and I've found that makes it easier to extend and debug as compared to the examples in the CUTLASS repo